meinberlin/react/livequestions/QuestionModerator.jsx

```diff
@@ -1,190 +1,189 @@
-import React from 'react'
+import React, { useState, useEffect, useCallback } from 'react'
 import django from 'django'
 import LikeCard from './LikeCard'
 import { classNames } from '../contrib/helpers'
 
-export default class QuestionModerator extends React.Component {
-  constructor (props) {
-    super(props)
+const hiddenText = django.gettext('mark as hidden')
+const undoHiddenText = django.gettext('undo mark as hidden')
+const doneText = django.gettext('mark as done')
+const addLiveText = django.gettext('added to live list')
+const removeLiveText = django.gettext('remove from live list')
+const addBookmarkText = django.gettext('add bookmark')
+const removeBookmarkText = django.gettext('remove bookmark')
 
-    this.state = {
-      is_on_shortlist: this.props.is_on_shortlist,
-      is_live: this.props.is_live,
-      likes: this.props.likes.count,
-      session_like: this.props.likes.session_like,
-      is_hidden: this.props.is_hidden,
-      is_answered: this.props.is_answered
-    }
-  }
+const QuestionModerator = (props) => {
+  const [isOnShortlist, setisOnShortlist] = useState(props.is_on_shortlist)
+  const [isLive, setIsLive] = useState(props.is_live)
+  const [isHidden, setIsHidden] = useState(props.is_hidden)
+  const [isAnswered, setIsAnswered] = useState(props.is_answered)
+  const [likeCount, setLikeCount] = useState(props.likes.count)
+  const [sessionLike, setSessionLike] = useState(props.likes.session_like)
+
+  useEffect(() => {
+    setisOnShortlist(props.is_on_shortlist)
+  }, [props.is_on_shortlist])
+
+  useEffect(() => {
+    setIsLive(props.is_live)
+  }, [props.is_live])
+
+  useEffect(() => {
+    setIsHidden(props.is_hidden)
+  }, [props.is_hidden])
+
+  useEffect(() => {
+    setIsAnswered(props.is_answered)
+  }, [props.is_answered])
```

Comment:
Why are you doing it this way instead of using a callback provided by the parent component? If you need to update your props, you should use a callback and not copy props into state.

Reply:
Ah, I see, you just converted the old code. Could you investigate it still? Only if it doesn't take much time. The live-questions aren't really used that much.

Reply:
> Why are you doing it this way instead of using a callback provided by the parent component?

@vellip I asked myself the same question of the original implementation, so I used all the `props` directly, without copying state, however doing this means on update of the parent component, everything is delayed due to polling, even with the changed `500` ms.

If we can think of an alternative approach, happy to change it!

Reply:
Eh, then let's leave it in. Maybe you can add a small comment with your findings or link to this discussion.

Reply:
Added a comment here: https://github.com/liqd/a4-meinberlin/pull/5751/files#diff-11e1fc49d7ab8e826f53c0b167259c0e8f36bff2da36beac044a81d4531cd694R39-R40